Mathlib/Probability/Process/BrownianMotion.lean

```diff
@@ -0,0 +1,80 @@
+/-
+Copyright (c) 2025 Peter Pfaffelhuber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Peter Pfaffelhuber
+-/
+
+import Mathlib.Analysis.InnerProductSpace.GramMatrix
+import Mathlib.MeasureTheory.Function.L2Space
+import Mathlib.MeasureTheory.Measure.Lebesgue.Basic
+
+/-! # Brownian Motion
+
+In this file we will eventually define Brownian Motion using the Kolmogorov extension theorem, and
+the Kolmogorov-Chentsov criterion for a continuous modification.
+
+At the moment, it only contains positive semidefiniteness of the covariance matrices for the
+finite-dimensional distributions.
+
+## Main definition
+
+## Main results
+
+* `Probability.BrownianMotion.posSemidef_covMatrix`:
+The matrix with `(s t : ℝ) ↦ s ∧ t` is positive semidefinite.
+
+-/
+
+section covariance
+
+open MeasureTheory NNReal Matrix Set
+
+open scoped ENNReal
+
+variable {n : Type*}
+variable {α : Type*} {mα : MeasurableSpace α} {μ : Measure α}
+
+namespace MeasureTheory
+
+namespace L2
+
+local notation "⟪" x ", " y "⟫" => @inner ℝ _ _ x y
+
+/-- In an `L2` space, the matrix of intersections of pairs of sets is positive semi-definite. -/
+theorem posSemidef_interMatrix [Fintype n] {μ : Measure α} {v : n → (Set α)}
+    (hv₁ : ∀ j, MeasurableSet (v j)) (hv₂ : ∀ j, μ (v j) ≠ ∞ := by finiteness) :
+    PosSemidef (of fun i j : n ↦ μ.real (v i ∩ v j)) := by
+  conv =>
+    right; right; intro i j;
+    rw [← id_eq (μ.real (v i ∩ v j)), ← RCLike.ofReal_real_eq_id,
+      ← inner_indicatorConstLp_one_indicatorConstLp_one (hv₁ i) (hv₁ j) (hv₂ i) (hv₂ j)]
```

Comment:
I think it would be reasonable to make a copy of `inner_indicatorConstLp_one_indicatorConstLp_one` for the case when `\bbk` is the reals, and the proof should be `:= inner_indicatorConstLp_one_indicatorConstLp_one`